src/hooks/useMaxSiblingWidth.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import Path from '../@types/Path'
+import editingValueStore from '../stores/editingValue'
+import parentOf from '../util/parentOf'
+
+/**
+ * Helper function to collect nodes with the specified `data-parent-path`.
+ * If no thought node is found, it continues searching with `data-grand-parent-path`.
+ */
+const collectNodes = (path: string): HTMLElement[] => {
+  /** Collect nodes by provided attribute and value. */
+  const collectByAttribute = (attribute: string, value: string): HTMLElement[] => {
+    return Array.from(document.querySelectorAll<HTMLElement>(`[data-${attribute}='${value}']`)).filter(
```

Comment:
Let's add a `tree-node` class name to the LayoutTree node to improve the performance of this attribute selector.